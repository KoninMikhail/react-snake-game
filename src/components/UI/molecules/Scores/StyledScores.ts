import styled from 'styled-components';
import { Text } from 'components/UI/atoms/body/Text/Text';
import { up } from 'styled-breakpoints';

export const StyledScores = styled.div`
  display: block;
`;

export const StyledScoreText = styled(Text)`
  font-size: 2.4em;

  ${up('xl')} {
    font-size: 3em;
  }
`;
