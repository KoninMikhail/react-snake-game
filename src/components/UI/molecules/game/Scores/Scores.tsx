import { StyledScores, StyledScoreText } from './StyledScores';
import { StyledGrid } from 'components/UI/extend/Grid/StyledGrid';
import { GridItem } from 'components/UI/extend/GridItem/GridItem';
import { Text } from 'components/UI/atoms/body/Text/Text';
import { TextAlign } from 'components/UI/atoms/body/TextAlign/TextAlign';
import padCharactersLeft from 'utils/padCharactersLeft';

/**
 * @typedef IScoresProps
 *
 * @property {("left" | "right" | "center")} align - Scores text alignment
 * @property  {number} score
 * @property  {number} maxScore
 */
interface IScoresProps {
  align: 'left' | 'right' | 'center';
  score: number;
  maxScore: number;
}

/**
 * Scores
 *
 * Use for display Game scores;
 *
 * @param {IScoresProps} props
 * @return {JSX.Element}
 * @constructor
 */
export const Scores = ({ align, score, maxScore }: IScoresProps): JSX.Element => {
  const isFormattedScore = padCharactersLeft(score, 6);

  return (
    <>
      <StyledScores>
        <StyledGrid gap="0">
          <GridItem>
            <TextAlign align={align}>
              <StyledScoreText as="span">{isFormattedScore}</StyledScoreText>
            </TextAlign>
          </GridItem>
          <GridItem>
            <TextAlign align={align}>
              <Text as="span">{'max-score: ' + maxScore}</Text>
            </TextAlign>
          </GridItem>
        </StyledGrid>
      </StyledScores>
    </>
  );
};
