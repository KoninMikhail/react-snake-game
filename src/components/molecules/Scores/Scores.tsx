import { StyledScores, StyledScoreText } from './StyledScores';
import { Grid } from 'components/extend/Grid/Grid';
import { GridItem } from 'components/extend/GridItem/GridItem';
import { Text } from 'components/atoms/body/Text/Text';
import { TextAlign } from 'components/atoms/body/TextAlign/TextAlign';
import padLeft from '../../../utils/padLeft';

/**
 * @typedef IScoresProps
 *
 * @property {("left" | "right" | "center")} align - Scores text alignment
 * @property  {number} score
 * @property  {number} maxScore
 */
interface IScoresProps {
  align: 'left' | 'right' | 'center';
  score: number;
  maxScore: number;
}

/**
 * Scores
 *
 * @description Use for display Game scores;
 *
 * @param {IScoresProps} props
 * @return {JSX.Element}
 * @constructor
 */

export const Scores = ({ align, score, maxScore }: IScoresProps): JSX.Element => {
  const formattedScore = padLeft(score, 6);

  return (
    <>
      <StyledScores>
        <Grid gap="0">
          <GridItem>
            <TextAlign align={align}>
              <StyledScoreText as="span">{formattedScore}</StyledScoreText>
            </TextAlign>
          </GridItem>
          <GridItem>
            <TextAlign align={align}>
              <Text as="span">{'max-score: ' + maxScore}</Text>
            </TextAlign>
          </GridItem>
        </Grid>
      </StyledScores>
    </>
  );
};
