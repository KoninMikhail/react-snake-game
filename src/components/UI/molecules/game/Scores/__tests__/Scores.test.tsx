import { render, screen } from '@testing-library/react';
import { Scores } from '../Scores';

describe('Scores', () => {
  it('render element with props', () => {
    render(<Scores align={'center'} score={2222} maxScore={15520} />);

    expect(screen.getByText(/002222/i)).toBeInTheDocument();
    expect(screen.getByText(/15520/i)).toBeInTheDocument();
  });

  it('render correctly', () => {
    render(<Scores align={'center'} score={2222} maxScore={15520} />);
    expect(screen.getByText(/002222/i)).toMatchSnapshot();
  });
});
