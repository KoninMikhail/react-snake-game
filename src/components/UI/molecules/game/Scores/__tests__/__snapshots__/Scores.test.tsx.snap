// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scores render correctly 1`] = `
.c0 {
  font-size: 2em;
}

@media (min-width:1200px) {

}

@media (min-width:1200px) {
  .c0 {
    font-size: 3em;
  }
}

<span
  class="c0"
>
  002222
</span>
`;
